cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -66,6 +66,50 @@ constexpr Unsigned to_unsigned(T signed_) {
   return static_cast<Unsigned>(signed_);
 }
 
+struct AbsoluteValue {
+  template <typename T, typename Arg>
+  static constexpr enable_if_floating_point<T> Call(KernelContext*, Arg arg, Status*) {
+    return (arg < static_cast<T>(0)) ? -arg : arg;
```

Comment:
Any reason don't use `std::fabs` directly?
std::fabs provides more optimized code: https://godbolt.org/z/nd38arvTe
std::fabs(-0.0) = 0.0, but `(x < 0) ? -x : x` outputs -0.0 if x = -0.0.

Reply:
I agree, thanks for the suggestion.